### ISSUE TITLE
Exception(Qonversion.offerings) - type 'Future<String?>' is not a subtype of type 'FutureOr<String>' in type cast

### ISSUE THREAD

**Issue by royston-c** [qonversion/flutter-sdk#69]
Somehow I can't believe I'm the first one to hit this so I might be doing something wrong since updating to the null-safe version of this package. Please let me know.

Calling for offerings `Qonversion.offerings();` throws a cast exception.

Package version: 3.0.0

**Error:**
```
type 'Future<String?>' is not a subtype of type 'FutureOr<String>' in type cast. 
Non-fatal Exception: FlutterError
0   ???                            0x0 offerings + 211 (qonversion.dart:211)
```

**Code to reproduce:**
```Dart
final offerings = await Qonversion.offerings();
```

**Code Location**
https://github.com/qonversion/flutter-sdk/blob/main/lib/src/qonversion.dart#L210-L211

I suspect this is also an issue:
https://github.com/qonversion/flutter-sdk/blob/main/lib/src/qonversion.dart#L223-224

**Comment by Maria-Bordunova** [comment]
Hi, @royston-c ! 
Thank you for the detailed report :raised_hands:. We'll check the scenario and come back in the near future.

**Comment by royston-c** [comment]
I just want to reiterate. This happens on EVER call to get offerings. This isn't an edge case.

Note: I tested the below and everything works:

```Dart
  static Future<QOfferings> offerings() async {
    final offeringsString = await _channel.invokeMethod<String>(Constants.mOfferings);

    // looks like the library returns an empty object from other methods
    if (offeringsString == null) {
       return QOfferings(null, <QOffering>[]);
    }

    final Map<String, dynamic> decodedOfferings = jsonDecode(offeringsString!);

    return QOfferings.fromJson(decodedOfferings);
  }
```

**Comment by Maria-Bordunova** [comment]
I see, thank you. We will fix it as soon as possible. I hope today the fix will be available:pray:

**Comment by royston-c** [comment]
I'm happy to create a PR myself if they'll be any delay. I was hoping to wrap up my null safety migration for my app today :)

**Comment by Maria-Bordunova** [comment]
Oh, it would be great! We really appreciate your contribution! :handshake:

**Comment by royston-c** [comment]
Sure. I'll get on that now.

**Comment by Maria-Bordunova** [comment]
@royston-c Thank you again. We published new release version 3.0.1 with your fixes :tada:

**Comment by royston-c** [comment]
thanks. I've just tested it and it seems to be fixed with 3.0.1 👍
